light_curves/code_src/rubin_functions.py

```diff
@@ -0,0 +1,210 @@
+from pathlib import Path
+import pyvo
+from astropy.table import Table, vstack, join
+import pandas as pd
+from data_structures import MultiIndexDFObject
+
+
+def rubin_authenticate():
+    """
+    Authenticate to the Rubin Science Platform TAP service using a locally stored token.
+
+    Returns
+    -------
+    rsp_tap : pyvo.dal.TAPService
+        An authenticated TAPService instance pointing to DP0 data, assigned to variable `rsp_tap`.
+        
+    Raises
+    ------
+    FileNotFoundError
+        If the token file does not exist or is empty.
+    RuntimeError
+        If the TAP session cannot be created or the returned baseurl mismatches the expected URL.
+    """
+    # Read token from home directory
+    token_file = Path.home() / '.rsp-tap.token'    
+    if not token_file.exists():
+        raise FileNotFoundError(f"Token file not found: {token_file}")
+    with open(token_file, 'r') as f:
+        token = f.readline().strip()
+    if not token:
+        raise ValueError(f"No token found in token file: {token_file}")
+
+    # Build credential and authenticated session
+    cred = pyvo.auth.CredentialStore()
+    cred.set_password("x-oauth-basic", token)
+    session = cred.get("ivo://ivoa.net/sso#BasicAA")
+    if session is None:
+        raise RuntimeError("Failed to obtain authenticated session. Check your token permissions.")
+
+    # Instantiate TAPService
+    rsp_tap_url = 'https://data.lsst.cloud/api/tap'
+    rsp_tap = pyvo.dal.TAPService(rsp_tap_url, session=session)
+    if rsp_tap.baseurl != rsp_tap_url:
+        raise RuntimeError(
+            f"rubin_authenticate: unexpected TAPService.baseurl "
+            f"'{rsp_tap.baseurl}', expected '{rsp_tap_url}'"
+        )
+    return rsp_tap
+
+
+    
+def rubin_get_objectids(sample_table, rsp_tap, search_radius=0.001):
+    """
+    Perform cone searches for each row in sample_table to retrieve matching DP0 object IDs.
+
+    Parameters
+    ----------
+    sample_table : astropy.table.Table
+        Table with columns ['coord', 'objectid', 'label'], where 'coord' is a SkyCoord.
+    rsp_tap : pyvo.dal.TAPService
+        Authenticated TAP service instance returned by `rubin_authenticate()`.
+    search_radius : float, deg
+        Radius of the cone search in degrees (default: 0.001).
+
+    Returns
+    -------
+    combined : astropy.table.Table
+        Combined table with columns ['coord_ra', 'coord_dec', 'objectId',
+        'in_objid', 'in_label'] tagging each DP0 Object match with the input sample info.
+    """
+    #could consider parallelizing this but for now we wait for TAP table upload and real data
+    all_tables = []
+    for row in sample_table:
```

Comment:
do they support table upload? that way you could avoid manually looping over the coordinates and have just one query.

Reply:
(this is for follow-up, easily point beyond the scope of this PR)

Reply:
I have requested table upload with RSP, Gregory knows this and told me this was a priority for them.  For now, it doesn't exist.